pkg/provider/doppler/client.go

```diff
@@ -74,16 +77,37 @@ type SecretsClientInterface interface {
 }
 
 func (c *Client) setAuth(ctx context.Context) error {
-	token, err := resolvers.SecretKeyRef(
-		ctx,
-		c.kube,
-		c.storeKind,
-		c.namespace,
-		&c.store.Auth.SecretRef.DopplerToken)
-	if err != nil {
-		return err
+	if c.store.Auth.SecretRef != nil {
```

Comment:
if both secretref is nil and oidcconfig is nil this will not return an error. is this really intended?